Add skip navigation link

derick-montague opened the pull request:
- Add watchers to change focus on ApplicationHeader component element.
Thisis needed because SPAs do not perform a full page load.

- Add styling for skip to content button

derick-montague committed:
Add skip navigation link

- Add watchers to change focus on ApplicationHeader component element.
Thisis needed because SPAs do not perform a full page load.

- Add styling for skip to content button

dixsie reviewed (commented):
✅ Tested in Chrome with screen reader

Skip link isn't working in Firefox or Safari

derick-montague commented:
Expected testing behavior. When you navigate between pages, you should be able to tab and have the `Skip to content` link transition on to the page. By hitting enter the link should transition off the screen and the next tab should be to the first focusable item on the page.

derick-montague commented:
> ✅ Tested in Chrome with screen reader
> 
> Skip link isn't working in Firefox or Safari

Update your System Preferences
https://www.weba11y.com/blog/2014/07/07/keyboard-navigation-in-mac-browsers/

yoshiemuranaka commented on src/layouts/AppLayout.vue:
```diff
@@ -24,6 +24,28 @@ export default {
+      // $nextTick = DOM updated
+      this.$nextTick(function() {
+        // set timeout needs a slight delay before referencing the DOM
+        setTimeout(() => {
```
settimeout 0 looks a little strange to me. Did you run into issues without it? Doesn't the $nextTick callback take care of DOM not being fully rendered?

derick-montague committed:
Remove setTimeout call

The nextTick method is sufficient. The setTimeout call was used to
handle an iOS 7 bug, which is not a device we support.

derick-montague commented:
You good now, @yoshiemuranaka?

derick-montague commented:
Integrated to webui-vue in https://gerrit.openbmc-project.xyz/c/openbmc/webui-vue/+/28610
